src/twisted/trial/_asynctest.py

```diff
@@ -81,36 +94,14 @@ def _eb(failure):
 
     failUnlessFailure = assertFailure
 
+    async def _runCorofnWithWarningsSuppressed(self, method):
+        with utils.suppressWarningsCM(self._getSuppress()):
+            v = method()
+            return (await v) if isinstance(v, Awaitable) else v
+
     def _run(self, methodName, result):
         from twisted.internet import reactor
 
-        timeout = self.getTimeout()
-
-        def onTimeout(d):
-            e = defer.TimeoutError(
-                f"{self!r} ({methodName}) still running at {timeout} secs"
-            )
-            f = failure.Failure(e)
-            # try to errback the deferred that the test returns (for no gorram
-            # reason) (see issue1005 and test_errorPropagation in
-            # test_deferred)
-            try:
-                d.errback(f)
-            except defer.AlreadyCalledError:
```

Comment:
Instead of just crashing here we keep passing the cancellation down the chain of Deferred https://github.com/twisted/twisted/blob/release-21.2.0-10091/src/twisted/internet/defer.py#L737